contract/models/contract_modification.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2020 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ContractModification(models.Model):
+
+    _name = 'contract.modification'
+    _description = 'Contract Modification'
+    _order = 'date desc'
+
+    date = fields.Date(
+        required=True,
+        string='Date'
+    )
+    description = fields.Text(
+        required=True,
+        string='Description'
+    )
+    contract_id = fields.Many2one(
+        string='Contract',
+        comodel_name='contract.contract',
+        required=True,
+        ondelete='cascade',
+        index=True
+    )
+    sent = fields.Boolean(
+        string='Mail notify',
+        default=False
+    )
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        record = super(ContractModification, self).create(vals_list)
+        record.check_modification_ids_need_sent()
+        return record
+
+    def write(self, vals):
+        res = super(ContractModification, self).write(vals)
+        self.check_modification_ids_need_sent()
+        return res
+
+    def check_modification_ids_need_sent(self):
+        records_not_sent = self.filtered(lambda x: not x.sent)
+        if records_not_sent:
+            records_not_sent.mapped('contract_id')._modification_mail_send()
```

Comment:
idk if it's possible, but mapped can return repeated items.

Reply:
@carlosdauden @pedrobaeza It's possible and need to improve `_modification_mail_send` function to use multi records?

Reply:
Char field mapped can return repeated items, but Many2one field mapped retuns a browse with unique items